hudi-common/src/main/java/org/apache/hudi/common/table/log/block/HoodieDataBlock.java

```diff
@@ -61,7 +61,7 @@ public abstract class HoodieDataBlock extends HoodieLogBlock {
 
   private final boolean enablePointLookups;
 
-  protected final Schema readerSchema;
+  protected Schema readerSchema;
```

Comment:
We should not be relaxing mutability constraints unless absolutely have to

Reply:
@wzx140 we'd need to revert these changes after we rebase on the latest master (these issues should have already been addressed on master)